src/Microsoft.ML.LightGbm/LightGbmArguments.cs

```diff
@@ -1,678 +1,351 @@
-            SortOrder = 2, ShortName = "mil", NullName = "<Auto>")]
-        [TGUI(Label = "Min Documents In Leaves", SuggestedSweeps = "1,10,20,50 ")]
-        [TlcModule.SweepableDiscreteParamAttribute("MinDataPerLeaf", new object[] { 1, 10, 20, 50 })]
-        public int? MinimumExampleCountPerLeaf;
-
-        /// <summary>
-        /// The maximum number of bins that feature values will be bucketed in.
-        /// </summary>
-        /// <remarks>
-        /// The small number of bins may reduce training accuracy but may increase general power (deal with over-fitting).
-        /// </remarks>
-        [Argument(ArgumentType.AtMostOnce, HelpText = "Maximum number of bucket bin for features.", ShortName = "mb")]
-        public int MaximumBinCountPerFeature = 255;
+        private protected OptionsBase BoosterOptions;
+    }
 
-        /// <summary>
-        /// Determines which booster to use.
-        /// </summary>
-        /// <value>
-        /// Available boosters are <see cref="DartBooster"/>, <see cref="GossBooster"/>, and <see cref="TreeBooster"/>.
-        /// </value>
-        [Argument(ArgumentType.Multiple, HelpText = "Which booster to use, can be gbtree, gblinear or dart. gbtree and dart use tree based model while gblinear uses linear function.", SortOrder = 3)]
-        public ISupportBoosterParameterFactory Booster = new TreeBooster.Options();
+    /// <summary>
+    /// Gradient boosting decision tree.
+    /// </summary>
+    /// <remarks>
+    /// For details, please see <a href="https://en.wikipedia.org/wiki/Gradient_boosting#Gradient_tree_boosting">gradient tree boosting</a>.
+    /// </remarks>
+    public sealed class GradientBooster : BoosterParameterBase
+    {
+        internal const string Name = "gbdt";
+        internal const string FriendlyName = "Tree Booster";
 
         /// <summary>
-        /// Determines whether to output progress status during training and evaluation.
+        /// The options for <see cref="GradientBooster"/>, used for setting <see cref="Booster"/>.
         /// </summary>
-        [Argument(ArgumentType.AtMostOnce, HelpText = "Verbose", ShortName = "v")]
-        public bool Verbose = false;
+        [TlcModule.Component(Name = Name, FriendlyName = FriendlyName, Desc = "Traditional Gradient Boosting Decision Tree.")]
+        public sealed class Options : OptionsBase
+        {
+            internal override BoosterParameterBase BuildOptions() => new GradientBooster(this);
+        }
 
-        /// <summary>
-        /// Controls the logging level in LighGBM.
-        /// </summary>
-        /// <value>
-        /// <see langword="true"/> means only output Fatal errors. <see langword="false"/> means output Fatal, Warning, and Info level messages.
-        /// </value>
-        [Argument(ArgumentType.AtMostOnce, HelpText = "Printing running messages.")]
-        public bool Silent = true;
+        public GradientBooster(Options options)
```

Comment:
>GradientBooster [](start = 15, length = 15)

`DartBooster` is internal

Reply:
So that is actually intentional since the user doesnt construct a DartBooster, they construct a DartBooster.Options. However, you do point out that GradientBooster is public -- so I made that internal.

---
In reply to: [266665204](https://github.com/dotnet/machinelearning/pull/2948#discussion_r266665204) [](ancestors = 266665204)